packages/babel-parser/src/parser/location.js

```diff
@@ -98,6 +98,9 @@ export const Errors = Object.freeze({
   MissingUnicodeEscape: "Expecting Unicode escape sequence \\uXXXX",
   MixingCoalesceWithLogical:
     "Nullish coalescing operator(??) requires parens when mixing with logical operators",
+  ModuleAttributesWithDuplicateKeys:
+    "Module attributes should not include duplicate keys",
+  ModuleAttributesWithoutType: "Module attributes should include a type",
```

Comment:
Instead of throwing if there isn't `type`, can we throw if there is anything other than `type`?

```suggestion
  ModuleAttributeDifferentFromType: "The only accepted module attribute is `type`",
```

Reply:
But we do allow other attributes too right? 
Like you can specify 
```
import file from "file.json". with loading: true, hasData: false
```
In this case the error should be `Module attributes should include a type`

Or are we restricting any other attribute other than `type` for now
@nicolo-ribaudo

Reply:
Ok, I thought that we discussed this but I can't even find the discussion where we decided to throw if `type` isn't provided :thinking:

cc @xtuc What should we allow for now?

Reply:
@nicolo-ribaudo I think that was a very initial discussion that we had when I had just started working on this PR.
I can revert the changes if the `type` key is optional as well after @xtuc confirms

Reply:
```suggestion
  ModuleAttributesWithoutType: "Module attributes should include a \"type\" key",
```
`Type` is a bit general and it may confuse TypeScript users.